lib/services/parser-service.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview ESLint Parser
+ * @author Nicholas C. Zakas
+ */
+/* eslint class-methods-use-this: off -- Anticipate future constructor arguments. */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Types
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("../linter/vfile.js").VFile} VFile */
+/** @typedef {import("@eslint/core").Language} Language */
+/** @typedef {import("@eslint/core").LanguageOptions} LanguageOptions */
+
+//-----------------------------------------------------------------------------
+// Exports
+//-----------------------------------------------------------------------------
+
+/**
+ * The parser for ESLint.
+ */
+class ParserService {
+
+    /**
+     * Parses the given file synchronously.
+     * @param {VFile} file The file to parse.
+     * @param {{language:Language,languageOptions:LanguageOptions}} config The configuration to use.
+     * @returns {Object} An object with the parsed source code or errors.
+     * @throws {Error} If the parser returns a promise.
+     */
+    parseSync(file, config) {
+
+        const { language, languageOptions } = config;
+        const result = language.parse(file, { languageOptions });
+
+        if (result.then) {
```

Comment:
```suggestion
        if (typeof result.then === "function") {
```

This is a common way to check if something is a promise/thenable.